src/app/models/proptypes.js

```diff
@@ -1,29 +1,60 @@
-import { arrayOf, shape, string, bool } from 'prop-types';
+import { arrayOf, bool, oneOfType, shape, string } from 'prop-types';
 
-export const imagePropTypes = {
-  model: shape({
-    blocks: arrayOf(
-      shape({
-        locator: string,
-      }),
-    ),
-  }),
+export const textPropTypes = {
+  blocks: arrayOf(
+    shape({
+      model: shape({
+        blocks: arrayOf(
+          shape({
+            model: shape({
+              text: string.isRequired,
+            }).isRequired,
+          }).isRequired,
+        ).isRequired,
+      }).isRequired,
+    }).isRequired,
+  ).isRequired,
 };
 
-export const textPropTypes = {
+export const optionalTextPropTypes = {
   blocks: arrayOf(
     shape({
       model: shape({
         blocks: arrayOf(
           shape({
-            text: string,
+            model: shape({
+              text: string,
+            }),
           }),
         ),
       }),
     }),
   ),
 };
 
+export const imagePropTypes = {
+  model: shape({
+    blocks: arrayOf(
+      oneOfType([
+        // rawImage block
+        shape({
+          model: shape({
+            locator: string.isRequired,
+          }),
+        }).isRequired,
+        // altText block
+        shape({
+          model: shape(textPropTypes),
+        }),
+        // caption block
+        shape({
+          model: shape(optionalTextPropTypes),
+        }).isRequired,
```

Comment:
Why is this block required if the internal propTypes are optional?

Reply:
Thanks for catching this. Now fixed.